src/libraries/System.Drawing.Common/src/System/Drawing/ImageInfo.cs

```diff
@@ -66,7 +67,21 @@ public ImageInfo(Image image)
                             }
 
                             // Frame delays are stored in 1/100ths of a second; convert to milliseconds while accumulating
-                            _frameEndTimes[f] = (lastEndTime += (BitConverter.ToInt32(values, i) * 10));
+                            // Per spec, a frame delay can be 0 which is treated as a single animation tick
+                            int delay = BitConverter.ToInt32(values, i) * 10;
+                            lastEndTime += delay > 0 ? delay : ImageAnimator.AnimationDelayMS;
+
+                            // Guard against overflows
+                            if (lastEndTime < _totalAnimationTime)
```

Comment:
this is for cases where delay is negative? does that ever happen/is it legal or are you just being careful?

Reply:
if just being careful, could you instead simply force delay to be no less than zero? then I think you wouldn't need _totalAnimationTime?

Reply:
I'm being careful mostly. Here's how it comes together though.

1. If the delay is <= 0, then apply the default animation delay
2. If the accumulated last end time overflows, then collapse all of the remaining frames into that instant (essentially skipping any frames beyond the overflow of animation time)
3. Ensure that TotalAnimationTime is a non-zero value if any frames are to be animated
4. Eliminate the array access from TotalAnimationTime, replacing that with _totalAnimationTime